apps/src/componentLibrary/tabs/_Tab.tsx

```diff
@@ -67,30 +90,39 @@ const _Tab: React.FunctionComponent<TabsProps> = ({
   const handleClick = useCallback(() => onClick(value), [onClick, value]);
   checkTabForErrors(isIconOnly, icon, text);
 
+  const buttonContent = renderButtonContent(
+    isIconOnly,
+    icon,
+    text,
+    iconLeft,
+    iconRight
+  );
+
+  const buttonElement = (
+    <button
+      type="button"
+      role="tab"
+      aria-selected={isSelected}
+      aria-controls={tabPanelId}
+      id={tabButtonId}
+      className={classNames(
+        isSelected && moduleStyles.selectedTab,
+        isIconOnly && moduleStyles.iconOnlyTab
+      )}
+      onClick={handleClick}
+      disabled={disabled}
+    >
+      {buttonContent}
+    </button>
+  );
+
   return (
     <li role="presentation">
-      <button
-        type="button"
-        role="tab"
-        aria-selected={isSelected}
-        aria-controls={tabPanelId}
-        id={tabButtonId}
-        className={classNames(
-          isSelected && moduleStyles.selectedTab,
-          isIconOnly && moduleStyles.iconOnlyTab
-        )}
-        onClick={handleClick}
-        disabled={disabled}
-      >
-        {isIconOnly && icon && <FontAwesomeV6Icon {...icon} />}
-        {!isIconOnly && (
-          <>
-            {iconLeft && <FontAwesomeV6Icon {...iconLeft} />}
-            {text && <span>{text}</span>}
-            {iconRight && <FontAwesomeV6Icon {...iconRight} />}
-          </>
-        )}
-      </button>
+      {tooltip ? (
+        <WithTooltip tooltipProps={tooltip}>{buttonElement}</WithTooltip>
```

Comment:
Would it be possible to update `Tooltip` so that if the Tab has an icon, the tooltip caret could be set to the right (if iconRight) or left (if iconLeft)?

Here's what the tab current looks like with the tooltip:
<img width="128" alt="tooltip-center" src="https://github.com/code-dot-org/code-dot-org/assets/107423305/fc519f7e-287b-4e1b-b74d-c9b116ac9dc6">

And am wondering if we could move the caret if there's an icon:
<img width="268" alt="Screenshot 2024-06-20 at 2 47 44 PM" src="https://github.com/code-dot-org/code-dot-org/assets/107423305/dc53a829-5d1e-4b01-8853-ae7f992eaccf">

@markabarrett - curious what you think about this. Thanks!

Reply:
In this case, the icon and the label do not have the same meaning but in other cases I imagine they will. So could we add a way to set the position of tooltip caret regardless of whether there is can icon or not.

Reply:
Yeah, right now it's not possible. And I don't think it's a good idea to add tooltip directly to an icon.
However, there's a native support of 'native tooltips' for icons which might be the best and the most accessible option: Just pass `title` prop to the needed `icon`, and it'll work in the following way. (I actually think this `title` prop should be passed in most of icons usage)

Reply:
After discussion with you and Mark, we've decided to take a different approach with Tabs and the lock icon which indicates 'view only'. Thanks so much for the information about the native support of native tooltips!